### PR TITLE
Fixes to available-prefixes and available-ips API endpoints

### LINKED ISSUE #5832: API version 2.0/2.1 not honored for `available-prefixes` endpoint in v2.2
### Environment
* Nautobot version (Docker tag too if applicable): 2.2.5
* Python version: 3.11.9
* Database platform, version: 
* Middleware(s):

### Steps to Reproduce
1. Get the UUID of a prefix that can accommodate a child prefix
2. Send a POST request to the endpoint `/api/ipam/prefixes/{UUID}/available-prefixes/` to create a child prefix **with the API version of 2.0 or 2.1**

### Expected Behavior
I expect the JSON response to respond with a single `location` (or None) instead of a list of locations.

This is an example response from a v2.1 instance where you can see `'location': None,`:
```python
{'id': '8448f5d6-8188-4553-808d-5780545cb430',
 'object_type': 'ipam.prefix',
 'display': '192.0.2.0/30',
 'url': 'http://<REDACTED>/api/ipam/prefixes/8448f5d6-8188-4553-808d-5780545cb430/',
 'natural_slug': 'global_192-0-2-0_30_8448',
 'prefix': '192.0.2.0/30',
 'type': {'value': 'network', 'label': 'Network'},
 'network': '192.0.2.0',
 'broadcast': '192.0.2.3',
 'prefix_length': 30,
 'ip_version': 4,
 'date_allocated': None,
 'description': '',
 'status': {'id': '9b23bdb3-50a2-4368-bbe1-d714a6b764d8',
  'object_type': 'extras.status',
  'url': 'http://<REDACTED>/api/extras/statuses/9b23bdb3-50a2-4368-bbe1-d714a6b764d8/'},
 'role': None,
 'parent': {'id': '28cc5db3-6de6-470b-b14c-ece2bf4a634e',
  'object_type': 'ipam.prefix',
  'url': 'http://<REDACTED>/api/ipam/prefixes/28cc5db3-6de6-470b-b14c-ece2bf4a634e/'},
 'location': None,
 'namespace': {'id': '353598ed-47ae-4f21-95c6-ba5d58d51e67',
  'object_type': 'ipam.namespace',
  'url': 'http://<REDACTED>/api/ipam/namespaces/353598ed-47ae-4f21-95c6-ba5d58d51e67/'},
 'tenant': None,
 'vlan': None,
 'rir': None,
 'created': '2024-06-20T19:41:33.404569Z',
 'last_updated': '2024-06-20T19:41:33.404614Z',
 'tags': [],
 'notes_url': 'http://<REDACTED>/api/ipam/prefixes/8448f5d6-8188-4553-808d-5780545cb430/notes/',
 'custom_fields': {},
 'computed_fields': {},
 'relationships': {}}
```

### Observed Behavior
Instead the JSON object returned provides `locations` with a list.

Here is the response without a location being set:
```python
# Example code
headers = {
    "Content-Type": "application/json",
    "Authorization": "Token aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
    "Accept": "application/json; version=2.0",
}
requests.post(
    "https://demo.nautobot.com/api/ipam/prefixes/2105a867-82c6-448d-961d-91c1cb19a8ea/available-prefixes/",
    json={"prefix_length": 30, "status": "Active"},
    headers=headers,
).json()
```

```python
{'id': 'da6c6391-d5c9-49bf-b962-533cb381b458',
 'object_type': 'ipam.prefix',
 'display': '192.0.2.12/30',
 'url': 'https://demo.nautobot.com/api/ipam/prefixes/da6c6391-d5c9-49bf-b962-533cb381b458/',
 'natural_slug': 'global_192-0-2-12_30_da6c',
 'prefix': '192.0.2.12/30',
 'type': {'value': 'network', 'label': 'Network'},
 'network': '192.0.2.12',
 'broadcast': '192.0.2.15',
 'prefix_length': 30,
 'ip_version': 4,
 'date_allocated': None,
 'description': '',
 'status': {'id': '9f38bab4-4b47-4e77-b50c-fda62817b2db',
  'object_type': 'extras.status',
  'url': 'https://demo.nautobot.com/api/extras/statuses/9f38bab4-4b47-4e77-b50c-fda62817b2db/'},
 'role': None,
 'parent': {'id': '2105a867-82c6-448d-961d-91c1cb19a8ea',
  'object_type': 'ipam.prefix',
  'url': 'https://demo.nautobot.com/api/ipam/prefixes/2105a867-82c6-448d-961d-91c1cb19a8ea/'},
 'namespace': {'id': '733d191a-4067-4215-90a8-814bcfe28f03',
  'object_type': 'ipam.namespace',
  'url': 'https://demo.nautobot.com/api/ipam/namespaces/733d191a-4067-4215-90a8-814bcfe28f03/'},
 'tenant': None,
 'vlan': None,
 'rir': None,
 'locations': [],
 'created': '2024-06-20T19:11:14.400229Z',
 'last_updated': '2024-06-20T19:11:14.400256Z',
 'tags': [],
 'notes_url': 'https://demo.nautobot.com/api/ipam/prefixes/da6c6391-d5c9-49bf-b962-533cb381b458/notes/',
 'custom_fields': {},
 'computed_fields': {},
 'relationships': {'prefix_circuit': {'id': '97a7089d-008d-4576-a238-a8cfb55f1387',
   'url': 'https://demo.nautobot.com/api/extras/relationships/97a7089d-008d-4576-a238-a8cfb55f1387/',
   'label': 'Prefix per Circuit',
   'type': 'one-to-one',
   'source': {'label': 'circuit',
    'object_type': 'circuits.circuit',
    'objects': []}}}}
```

And here is the response with a location provided in the payload:
```python
headers = {
    "Content-Type": "application/json",
    "Authorization": "Token aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
    "Accept": "application/json; version=2.0",
}
requests.post(
    "https://demo.nautobot.com/api/ipam/prefixes/2105a867-82c6-448d-961d-91c1cb19a8ea/available-prefixes/",
    json={
        "prefix_length": 30,
        "status": "Active",
        "location": {"name": "Jersey City", "parent__name": "networktocode"},
    },
    headers=headers,
).json()
```

```python
{'id': '888d0573-69f5-449d-98b1-3fedd8503649',
 'object_type': 'ipam.prefix',
 'display': '192.0.2.20/30',
 'url': 'https://demo.nautobot.com/api/ipam/prefixes/888d0573-69f5-449d-98b1-3fedd8503649/',
 'natural_slug': 'global_192-0-2-20_30_888d',
 'prefix': '192.0.2.20/30',
 'type': {'value': 'network', 'label': 'Network'},
 'network': '192.0.2.20',
 'broadcast': '192.0.2.23',
 'prefix_length': 30,
 'ip_version': 4,
 'date_allocated': None,
 'description': '',
 'status': {'id': '9f38bab4-4b47-4e77-b50c-fda62817b2db',
  'object_type': 'extras.status',
  'url': 'https://demo.nautobot.com/api/extras/statuses/9f38bab4-4b47-4e77-b50c-fda62817b2db/'},
 'role': None,
 'parent': {'id': '2105a867-82c6-448d-961d-91c1cb19a8ea',
  'object_type': 'ipam.prefix',
  'url': 'https://demo.nautobot.com/api/ipam/prefixes/2105a867-82c6-448d-961d-91c1cb19a8ea/'},
 'namespace': {'id': '733d191a-4067-4215-90a8-814bcfe28f03',
  'object_type': 'ipam.namespace',
  'url': 'https://demo.nautobot.com/api/ipam/namespaces/733d191a-4067-4215-90a8-814bcfe28f03/'},
 'tenant': None,
 'vlan': None,
 'rir': None,
 'locations': [{'id': '373a540e-6ad6-4a14-be7a-1580f368d0e1',
   'object_type': 'dcim.location',
   'url': 'https://demo.nautobot.com/api/dcim/locations/373a540e-6ad6-4a14-be7a-1580f368d0e1/'}],
 'created': '2024-06-20T19:48:11.063743Z',
 'last_updated': '2024-06-20T19:48:11.063765Z',
 'tags': [],
 'notes_url': 'https://demo.nautobot.com/api/ipam/prefixes/888d0573-69f5-449d-98b1-3fedd8503649/notes/',
 'custom_fields': {},
 'computed_fields': {},
 'relationships': {'prefix_circuit': {'id': '97a7089d-008d-4576-a238-a8cfb55f1387',
   'url': 'https://demo.nautobot.com/api/extras/relationships/97a7089d-008d-4576-a238-a8cfb55f1387/',
   'label': 'Prefix per Circuit',
   'type': 'one-to-one',
   'source': {'label': 'circuit',
    'object_type': 'circuits.circuit',
    'objects': []}}}}
```

_FYI: `/available-ips/` may have the same issue, but I didn't test it yet._

### PR DESCRIPTION
**Pull request by glennmatthews**
# Closes #5832 
# What's Changed

- Fix the response to `POST /api/ipam/prefixes/<id>/available-prefixes/` so that it uses the correct API-versioned PrefixSerializer instead of always using the post-2.2 serializer class.
- Fixes to the OpenAPI schema for `available-prefixes` and `available-ips` API endpoints.
  - One caveat - I'm not sure how to do a versioned serializer in `@extend_schema`, so the *documented* response serializer to `/available-prefixes/` is still the post-2.2 version, even when using the 2.1 or 2.0 API versions. The actual response content is correctly versioned though.

# Screenshots

## Available IPs

image

image

## Available Prefixes

image

image

image

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- n/a Documentation Updates (when adding/changing features)
- n/a Example App Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/ipam/api/serializers.py [hunk 1]
```diff
@@ -321,27 +321,30 @@ class Meta:
...
+    """
+    Input serializer for POST to /api/ipam/prefixes/<id>/available-prefixes/, i.e. allocating one or more sub-prefixes.
+
+    Since setting of multiple locations on create is not supported, this uses the legacy single-location option.
+    """
+
+    prefix_length = serializers.IntegerField(required=True)
+
+    class Meta(PrefixLegacySerializer.Meta):
+        fields = PrefixLegacySerializer.Meta.fields.copy()
```
Comment (itdependsnetworks) [inline nautobot/ipam/api/serializers.py:334]: I would guess not, but just double checking deep copy would not be more appropriate.

Comment (glennmatthews) [reply nautobot/ipam/api/serializers.py:334]: It's just an array of strings so I'm pretty sure copy is OK. Thanks for checking!

File: nautobot/ipam/api/views.py [hunk 2]
```diff
@@ -214,9 +216,9 @@ def available_prefixes(self, request, pk=None):
                 # Initialize the serializer with a list or a single object depending on what was requested
                 context = {"request": request, "depth": 0}
                 if isinstance(request.data, list):
-                    serializer = serializers.PrefixSerializer(data=requested_prefixes, many=True, context=context)
+                    serializer = self.get_serializer_class()(data=requested_prefixes, many=True, context=context)
                 else:
-                    serializer = serializers.PrefixSerializer(data=requested_prefixes[0], context=context)
+                    serializer = self.get_serializer_class()(data=requested_prefixes[0], context=context)
```
Comment (glennmatthews) [inline nautobot/ipam/api/views.py:221]: This set of changes fixes the API versioning.

**Review by smk4664 — APPROVED** [review]
This looks good to me.

**Pull request merged**